tests/parser/features/external_contracts/test_external_contract_calls.py

```diff
@@ -222,7 +222,640 @@ def get_array(arg1: address) -> (Bytes[30], int128, Bytes[3]):
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> (int128, Bytes[3], int128): view
+
+@external
+def bar(arg1: address) -> (int128, Bytes[3], int128):
+    a: int128 = 0
+    b: Bytes[3] = b""
+    c: int128 = 0
+    a, b, c = Foo(arg1).foo()
+    return a, b, c
+"""
+
+    c2 = get_contract(contract_2)
+    assert c.foo() == [int(f"{(2**(a-1))-1}"), b"dog", int(f"{(2**(b-1))-1}")]
+    assert_tx_failed(lambda: c2.bar(c.address))
+
+
+@pytest.mark.parametrize("a,b", [(128, 256), (256, 128)])
+def test_tuple_with_int128_too_long_two(get_contract, assert_tx_failed, a, b):
+    contract_1 = f"""
+@external
+def foo() -> (int{b}, Bytes[3], int{a}):
+    return {(2**(b-1))-1}, b'dog', {(2**(a-1))-1}
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = f"""
+interface Foo:
+    def foo() -> (int{a}, Bytes[3], int{b}): view
+
+@external
+def bar(arg1: address) -> (int{a}, Bytes[3], int{b}):
+    a: int{a} = 0
+    b: Bytes[3] = b""
+    c: int{b} = 0
+    a, b, c = Foo(arg1).foo()
+    return a, b, c
+"""
+
+    c2 = get_contract(contract_2)
+    assert c.foo() == [int(f"{(2**(b-1))-1}"), b"dog", int(f"{(2**(a-1))-1}")]
+    assert_tx_failed(lambda: c2.bar(c.address))
+
+
+@pytest.mark.parametrize("type", ["uint8", "uint256", "int128", "int256"])
+def test_external_contract_calls_with_decimal(get_contract, type):
+    contract_1 = f"""
+@external
+def foo() -> {type}:
+    return 1
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> decimal: view
+
+@external
+def bar(arg1: address) -> decimal:
+    return Foo(arg1).foo()
+"""
+
+    c2 = get_contract(contract_2)
+    assert float(c2.bar(c.address)) == 1e-10
+
+
+def test_decimal_too_long(get_contract, assert_tx_failed):
+    contract_1 = """
+@external
+def foo() -> uint256:
+    return 2**255
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> decimal: view
+
+@external
+def bar(arg1: address) -> decimal:
+    return Foo(arg1).foo()
+"""
+
+    c2 = get_contract(contract_2)
+    assert_tx_failed(lambda: c2.bar(c.address))
+
+
+@pytest.mark.parametrize("a", ["uint8", "uint256", "int128", "int256"])
+@pytest.mark.parametrize("b", ["uint8", "uint256", "int128", "int256"])
+def test_tuple_with_decimal(get_contract, a, b):
+    contract_1 = f"""
+@external
+def foo() -> ({a}, Bytes[3], {b}):
+    return 0, b'dog', 1
+    """
+
+    c = get_contract(contract_1)
+
+    contract_2 = """
+interface Foo:
+    def foo() -> (decimal, Bytes[3], decimal): view
+
+@external
+def bar(arg1: address) -> (decimal, Bytes[3], decimal):
+    a: decimal = 0.0
+    b: Bytes[3] = b""
+    c: decimal = 0.0
+    a, b, c = Foo(arg1).foo()
+    return a, b, c
+"""
+
+    c2 = get_contract(contract_2)
+    assert c.foo() == [0, b"dog", 1]
+    result = c2.bar(c.address)
+    assert len(result) == 3
+    assert float(result[0]) == 0.0
```

Comment:
Why not use Decimal comparisons here? Something like
```
    assert result == (Decimal(0), b"dog", Decimal("1e-10"))
```

Reply:
Agree, thanks for the catch!